File: WebTool-Web/src/main/java/com.lxl.webtool/dao/BhImgquestionService.java
package com.lxl.webtool.dao;

import com.lxl.webtool.dao.pojo.TbBhImgquestion;
import com.lxl.webtool.pojo.BaseResult;
import com.lxl.webtool.pojo.PageResult;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface BhImgquestionService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbBhImgquestion> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(TbBhImgquestion bhImgquestion);


    /**
     * 修改
     */
    public void update(TbBhImgquestion bhImgquestion);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbBhImgquestion findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbBhImgquestion bhImgquestion, int pageNum, int pageSize);

    /**
     * 根据渠道和guid获取唯一标识
     *
     * @param channelkey
     * @param guid
     */
    public BaseResult findImgData(String channelkey, String guid);


    /**
     * 保存用户回复结果
     *
     * @param channelkey
     * @param guid
     * @param answser
     * @return
     */
    public BaseResult saveImgAnswer(String channelkey, String guid, String answser);

}

File: WebTool-Web/src/main/webapp/js/service/bhImgquestionService.js
//服务层
app.service('bhImgquestionService', function ($http) {

        //读取列表数据绑定到表单中
        this.findAll = function () {
            return $http.get('../puapi/bhImgquestion/findAll.do');
        }
        //分页
        this.findPage = function (page, rows) {
            return $http.get('../puapi/bhImgquestion/findPage.do?page=' + page + '&rows=' + rows);
        }
        //查询实体
        this.findOne = function (id) {
            return $http.get('../puapi/bhImgquestion/findOne.do?id=' + id);
        }
        //增加
        this.add = function (entity) {
            return $http.post('../puapi/bhImgquestion/add.do', entity);
        }
        //修改
        this.update = function (entity) {
            return $http.post('../puapi/bhImgquestion/update.do', entity);
        }
        //删除
        this.dele = function (ids) {
            return $http.get('../puapi/bhImgquestion/delete.do?ids=' + ids);
        }
        //搜索
        this.search = function (page, rows, searchEntity) {
            return $http.post('../puapi/bhImgquestion/search.do?page=' + page + "&rows=" + rows, searchEntity);
        }

        this.findImgData = function (channelkey, guid) {
            return $http.get('../puapi/bhImgquestion/findImgData.do?channelkey=' + channelkey + "&guid=" + guid);
        }

        this.saveImgAnswer = function (channelkey, guid, answser) {
            return $http.get('../puapi/bhImgquestion/saveImgAnswer.do?channelkey=' + channelkey + "&guid=" + guid + "&answser=" + answser);
        }
    }
);
